### ISSUE TITLE
Options for notifications

### ISSUE THREAD

**Issue by bb1a2** [aj3423/SpamBlocker#3]
I would like to have an option to disable notifications for each of:
1. pass SMS
2. blocked SMS

For example, there is no need for SpamBlocker to tell me about a passing message in a notification since my regular SMS app already does that. I currently get duplicate notifications.

For blocked notifications, it should be **per rule** setting. Some blocked messages are actual spam that I don't want to see. Others are just not important (e.g. remaining balance) that I don't want to be notified about but I still want them to be there in case I need them. Those last ones I may keep as silent notifications.

In summary:
Pass: no notification (global setting)
Spam: no notification (per rule)
Less important: silent notification (per rule)

**Comment by aj3423** [comment]
You need to disable the notifications of the regular SMS app, let this app take over the notifications, as mentioned in the readme, but I forgot to mention this on fdroid page.

Being able to set different notification type for each filter, this is a good idea, I'll try to implement.

**Comment by bb1a2** [comment]
>let this app take over the notifications

There is a problem with that. If you open the SMS, the SpamBlocker notification is not dismissed. I think SB doesn't know that the SMS was opened.

Besides, the regular app has some nice features like call/reply. I like your app, but I think the user should have the option.

**Comment by aj3423** [comment]
Indeed. I always launch SB first, by clicking the notification. When SB is launched, the notifications are dismissed. Then I click the bottom Call/SMS button to launch the regular app. A bit complex, but only takes about one second. 

About the features like call/reply,  currently there's no way to do that. I think it's necessary to disable the notification of the stock SMS app, it doesn't filter contents, it shows sound notification even for spam.

**Comment by bb1a2** [comment]
I see. I thought that Android (allowing for a separate spam blocker app) would allow SB somehow to tell the main SMS app: "this is spam" and the main SMS app wouldn't notify 😕

**Comment by aj3423** [comment]
Yeah, there is CallScreeningService for Call, which does what you said, but there is no such built-in service for SMS. Many people build their own SMS app just for filtering, that's a lot of work to do, and they all look ugly:) Actually they don't block messages, they just put them in a separate folder, hidden by default, only the notification is blocked, and this is what SB does, the main point to me is to not get disrupted, but there's no way to hide them, it's a trade-off.

**Comment by bb1a2** [comment]
OK. I can do it partly. Most of my spam actually comes from the telco itself. I can disable their notifications in the main SMS app and let SB handle the notifications for them. But this requires a default notification toggle for SB (i.e. don't notify by default for all numbers). I can make that work with a global rule inside SB once per rule notifications are available.

**Comment by aj3423** [comment]
Done, try the nightly release:
https://github.com/aj3423/SpamBlocker/actions/runs/8820222492 

https://github.com/aj3423/SpamBlocker/assets/4710875/f06faef6-e2c3-4207-9839-bebc8a608f87

**Comment by bb1a2** [comment]
Looks beautiful.
